odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/Odnoklassniki.java

```diff
@@ -124,18 +125,17 @@ private Odnoklassniki(final Context context, final String appId, final String ap
      * @param scopes      {@link OkScope} - application request permissions as per {@link OkScope}.
      * @see OkAuthType
      */
-    public final void requestAuthorization(OkListener listener, @Nullable String redirectUri,
+    public final void requestAuthorization(Activity activity, OkListener listener, @Nullable String redirectUri,
```

Comment:
But this depends on Odnoklassniki app's code. I don't have access to it's login activity to set result there. So listener will stay here.